src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/SplashScreen.cs

```diff
@@ -285,144 +293,145 @@ private void Fadeout_Tick(object unused, EventArgs args)
             }
         }
 
-        private void DestroyResources()
+        ~SplashScreen() => DestroyResources(finalizer: true);
```

Comment:
❔ Can we wrap unmanaged resources in the correct handle type and remove this finalizer?

Reply:
Deliberately not doing this. There is a non-trivial cost to using SafeHandle that will have measurable impact on WPF. In most cases things will be scoped to the stack so taking the hit of SafeHandle for API definitions is counterproductive in an interop heavy environment.